### PR TITLE
Added automatic screenshot capabilities

### PR DESCRIPTION
**Pull request by tclaus**
By entering $:fastlane snapshot
Screenshots can then copied to AppStore
TODO: Multilanguage Screenshots, analze need for automatic upload
Simulators must be prepared: Start the simulator and do a full login. Accept authorization process with a developer account. 
This is needed because of the system-privacy alerts which can't be overridden by code.

### CODE REVIEW

File: ios/insporationUITests/insporationUITests.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    func testTakeSnapshots() throws {
+        // Assumtions:
+        // App is prepaired with an logged-in account, first tab (Main) is selected and Streams is selected
+  
+        let app = XCUIApplication()
+        app.launch()
+        setupSnapshot(app)
+        sleep(3) // Wait until stream is loaded
```
Comment (jhass) [inline ios/insporationUITests/insporationUITests.swift]: Seems brittle, is there no way to wait until some specific element appears? We maybe could even just attach a semantic label to the loading spinner and wait for it to disappear?

File: ios/insporationUITests/insporationUITests.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        
+        snapshot("01FirstScreen")
+        
+        app.buttons["StreamType.main"].tap()
+        snapshot("02Selector")
+        app.otherElements["StreamType.followedTags"].tap()
+        sleep(2) // Wait until tags are loaded
+        snapshot("03Tags")
+        app.otherElements["Notifications\nNotifications\nTab 4 of 5"].tap()
```
Comment (jhass) [inline ios/insporationUITests/insporationUITests.swift:39]: Maybe wouldn't hurt to use semantic labels here too? :)

File: lib/stream_page.dart [hunk 3]
```diff
@@ -146,7 +146,7 @@ class _StreamTypeSelector extends StatelessWidget with LocalizationHelpers {
           },
           items: const [StreamType.main, StreamType.activity, StreamType.aspects, StreamType.followedTags,
             StreamType.mentions, StreamType.liked, StreamType.commented].map((type) =>
-              DropdownMenuItem(child: Text(l(context).streamName(type)), value: type)).toList()
+              DropdownMenuItem(child: Text(l(context).streamName(type),semanticsLabel: type.toString()), value: type)).toList()
```
Comment (jhass) [inline lib/stream_page.dart:152]:
```suggestion
              DropdownMenuItem(child: Text(l(context).streamName(type), semanticsLabel: type.toString()), value: type)).toList()
```

**Review by jhass — APPROVED** [review]
I feel like it could be made slightly more robust, but nothing major :)

File: ios/insporationUITests/insporationUITests.swift (hunk 2 above)
Comment (tclaus) [reply ios/insporationUITests/insporationUITests.swift:39]: I found out that NavigationButtonItems in flutter doesn't have any semantics. It seems that visible text and keys is divided by \n . 
It turns out that after selecting a StreamType - button, it's not reliable to go back.

Comment (jhass) [reply ios/insporationUITests/insporationUITests.swift:39]: Did you try to set the `semanticLabel` on the `Icon`? https://github.com/jhass/insporation/blob/e574c7eba6ae6435e435dcdedad1bd69fbbfef9e/lib/src/navigation.dart#L33

Comment (jhass) [reply ios/insporationUITests/insporationUITests.swift:39]: Otoh it seems this is used by TalkOver and VoiceOver, so for blind users. Maybe we shouldn't misappropriate it for UI tests :/

https://api.flutter.dev/flutter/semantics/SemanticsConfiguration/label.html

Comment (jhass) [reply ios/insporationUITests/insporationUITests.swift:39]: Does setting `key: Key("foo")` to stuff turn up in the widget tree visible to the UI tests?

Comment (jhass) [reply ios/insporationUITests/insporationUITests.swift:39]: This is how flutter level integration tests find their widgets at least: https://flutter.dev/docs/cookbook/testing/integration/introduction

**Comment by weex** [comment]
Bump on conflicts

**Pull request closed without merging**